components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java

```diff
@@ -221,4 +215,87 @@ private static void registerExtensionValidators(ClassLoader classLoader) {
             return super.readWSDLDocument(domModel, file, messagegenerator, wsdlvalinfo);
         }
     }
+
+    /**
+     * Load XML data to a temporary file.
+     *
+     * @param document XML DOM
+     * @return URL of the file
+     * @throws IOException          on error writing to file
+     * @throws TransformerException on transforming error
+     */
+    private URL loadXMLToFile(Document document) throws TransformerException, IOException {
+
+
+        DOMSource source = new DOMSource(document);
+        File tempFile = File.createTempFile("temp", ".txt");
```

Comment:
Is this method thread safe? Is it possible for multiple threads to invoke this method at the same time? If so the temp file will get overwritten by other threads.